joern-cli/frontends/rubysrc2cpg/src/main/scala/io/joern/rubysrc2cpg/passes/ImportsPass.scala

```diff
@@ -0,0 +1,26 @@
+package io.joern.rubysrc2cpg.passes
+
+import io.shiftleft.codepropertygraph.Cpg
+import io.shiftleft.codepropertygraph.generated.nodes.*
+import io.shiftleft.semanticcpg.language.*
+import io.shiftleft.passes.ConcurrentWriterCpgPass
+import io.joern.x2cpg.Imports.createImportNodeAndLink
+import io.joern.x2cpg.X2Cpg.stripQuotes
+
+/** Derived from XImportsPass TODO: maybe make XImportsPass more generic
+  */
+class ImportsPass(cpg: Cpg) extends ConcurrentWriterCpgPass[Call](cpg) {
+  protected val importCallName: String = "require"
+  override def generateParts(): Array[Call] = cpg
+    .call(importCallName)
+    .toArray
+
+  override def runOnPart(diffGraph: DiffGraphBuilder, call: Call): Unit = {
+    val importedEntity = stripQuotes(call.argument.isLiteral.code.l match {
+      case s :: _ => s
+      case _      => ""
+    })
+    val importedAs = ""
```

Comment:
`importedAs` == `importedEntity`. When that is not the case, then it is usually imported under some alias, e.g.

`import foo as bar` or, as it's used in the Java world `import bar.Foo` can be accessed as `Foo`.